### PR TITLE
Refactor logit transform logic into TK

### PR DESCRIPTION
**Pull request by dmcdougall**
The fact we were doing logit transforms outside of the TK designed for logit transforms bothered me.

I can't do it everywhere, though, one of the pieces of logic relies fundamentally on the adaptivity formulation.  See #555 for details.

### CODE REVIEW

File: src/stats/inc/TKGroup.h [hunk 1]
```diff
@@ -115,6 +115,14 @@ class BaseTKGroup {
...
    */
   virtual void cleanCovMatrix() = 0;
+
+  //! Sets and proposal covariance matrix (and for all DR stage RVs as well)
+  /*!
+   * If the TK has a proposal covariance matrix, set the covariance matrix.
+   * If the TK support delayed rejection, derived classes should scale by a
+   * factor of \f$ 1/scales^2 \f$.
+   */
+  virtual void updateLawCovMatrix(const M & covMatrix) = 0;
```
Comment (dmcdougall) [inline src/stats/inc/TKGroup.h]: Note, this change break backwards compatibility.  In my defence, I highly doubt anybody is subclassing TKs yet; the factory infrastructure is quite new.

Instead, I can make this function virtual but non-pure and throw a run-time exception instead.

**Comment by dmcdougall** [comment]
This PR serves as a first step in addressing #472.

File: src/stats/inc/TKGroup.h (hunk 1 above)
Comment (roystgnr) [inline src/stats/inc/TKGroup.h]: I'd definitely prefer the compile-time failure option to the run-time failure option...

**Comment by roystgnr** [comment]
:+1:

**Pull request merged**